pkg/kfapp/ossm/k8s_resources.go

```diff
@@ -168,3 +168,95 @@ func (o *OssmInstaller) PatchResourceFromFile(filename string, elems ...configty
 	}
 	return nil
 }
+
+func (o *OssmInstaller) CheckForCRD(group string, version string, resource string) error {
+	dynamicClient, err := dynamic.NewForConfig(o.config)
+	if err != nil {
+		log.Error(err, "Failed to initialize dynamic client")
+	}
+
+	crdGVR := schema.GroupVersionResource{
+		Group:    group,
+		Version:  version,
+		Resource: resource,
+	}
+
+	_, err = dynamicClient.Resource(crdGVR).List(context.Background(), metav1.ListOptions{})
+	return err
```

Comment:
Is it possible that the list returned is empty but error is `nil`? In other words - if CRD is not installed does `.List` operation returns `NotFound` error or could it be possible that the list returned is empty?

Reply:
No, if CRD is not installed error will be thrown. List returned is empty but err is nil if the CRD is installed but there are no objects of said CR created. 

Tested this by deleting Authorino operator + CRD to ensure.

Reply:
Ah right, you are fetching the actual resource, not the definition :)